tools/metal-support/main.go

```diff
@@ -0,0 +1,989 @@
+			return m, tea.Batch(waitForFetch(m.sub), cmd)
+		}
+
+	case fetchSuccessMessage:
+		m.versions = msg.versions
+		m.isLoadingInitialData = false
+		m.fetching = false
+		m.lastUpdated = time.Now()
+
+		err := Serialize(Cache{
+			OCPVersions: m.versions,
+			LastUpdated: m.lastUpdated,
+		}, m.cachePath)
+
+		if err != nil {
+			m.err = fmt.Errorf("failed to serialize data: %w", err)
+		}
+
+		return m, nil
+	case tea.WindowSizeMsg:
+		m.width = msg.Width
+		m.height = msg.Height
+	case tea.KeyMsg:
+		switch {
+		case key.Matches(msg, m.keys.Quit):
+			return m, tea.Quit
+		case key.Matches(msg, m.keys.Up):
+			m.selectedJob--
+			if m.selectedJob < 0 {
+				m.selectedJob = 0
+			}
+		case key.Matches(msg, m.keys.Down):
+			m.selectedJob++
+			if m.selectedJob > len(m.versions[m.selectedVersion].Jobs)-1 {
+				m.selectedJob = len(m.versions[m.selectedVersion].Jobs) - 1
+			}
+		case key.Matches(msg, m.keys.Right):
+			m.selectedVersion++
+			if m.selectedVersion > len(m.versions)-1 {
+				m.selectedVersion = len(m.versions) - 1
+			}
+		case key.Matches(msg, m.keys.Left):
+			m.selectedVersion--
+			if m.selectedVersion < 0 {
+				m.selectedVersion = 0
+			}
+		case key.Matches(msg, m.keys.Open):
+			job := m.versions[m.selectedVersion].Jobs[m.selectedJob]
+			OpenInBrowser(job.HistoryLink())
+		case key.Matches(msg, m.keys.Update):
+			m.sub = make(chan float64, 10)
+			m.fetching = true
+			m.fetchProgress = newProgress()
+			return m, tea.Batch(
+				startRefreshJob(m.versions, m.selectedVersion, m.selectedJob, m.sub),
+				waitForFetch(m.sub),
+			)
+		case key.Matches(msg, m.keys.UpdateAll):
+			m.sub = make(chan float64, 10)
+			m.fetching = true
+			m.fetchProgress = newProgress()
+			return m, tea.Batch(
+				startRefreshAllJobs(m.versions, m.sub),
+				waitForFetch(m.sub),
+			)
+		case key.Matches(msg, m.keys.Help):
+			m.helpVisible = !m.helpVisible
+		}
+	case progress.FrameMsg:
+		if m.isLoadingInitialData {
+			progressModel, cmd := m.initialProgress.Update(msg)
+			m.initialProgress = progressModel.(progress.Model)
+			return m, cmd
+		}
+
+		if m.fetching {
+			progressModel, cmd := m.fetchProgress.Update(msg)
+			m.fetchProgress = progressModel.(progress.Model)
+			return m, cmd
+		}
+
+		return m, nil
+	}
+
+	return m, nil
+}
+
+func (m model) View() string {
+	if m.isLoadingInitialData {
+		s := "Loading data, please wait...\n"
+		s += m.initialProgress.View()
+		return lipgloss.NewStyle().Align(lipgloss.Center, lipgloss.Center).Width(m.width).Height(m.height - 2).Render(s)
+	}
+
+	if m.helpVisible {
+		return m.help.FullHelpView(m.keys.FullHelp())
+	}
+
+	blocks := []string{}
+
+	for i, version := range m.versions {
+		b := ""
+
+		if m.selectedVersion == i {
+			b += versionStyleSelected.Render(version.Name)
+		} else {
+			b += versionStyle.Render(version.Name)
+		}
+
+		for index, job := range version.Jobs {
+			b += "\n"
+
+			jobName := job.Name
+
+			if strings.HasPrefix(jobName, "metal-ipi-") {
```

Comment:
I'd love to use it also for the agent jobs. What about specifying the job prefixes as env var or config setting?